File: LeetCode Solutions/Top Interview Questions(Easy Collection)/Linked List/Reverse Linked List/Reverse Linked List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* reverseList(struct ListNode* head){
    struct ListNode *prev = NULL;
    struct ListNode *curr = head ;     
    while (curr != NULL){
        head = curr;
        curr = curr->next;
        head->next = prev;
        prev = head;
    }
    return head;
}

File: LeetCode Solutions/Linked List/Reverse Linked List/Reverse Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        prev = None
        curr = head      
        while (curr):
            head = curr
            curr = curr.next
            head.next = prev
            prev = head
        return head